source/magpie/dataframe.d

```diff
@@ -1773,6 +1778,52 @@ public:
         ret.indx.generateCodes();
         return ret;
     }
+
+    /++
+    Filter the DataFrame based on the result of callback
+    Params:
+        filterFunc: Callback function based on whose result, the DataFrame is filtered
+    Returns:
+        Filtered DataFrame
+    +/
+    auto filter(alias filterFunc)() @property
```

Comment:
is there any particular reason why you can't use https://dlang.org/phobos/std_algorithm_iteration.html#.filter ?

Reply:
It was mainly due to difference in data type of Homogeneous and Heterogeneous DataFrame.